stan/math/mix/functor/laplace_marginal_density.hpp

```diff
@@ -45,11 +45,21 @@ struct laplace_options {
   double tolerance{1e-6};
   /* Maximum number of steps*/
   int max_num_steps{100};
+};
+
+template <typename Theta, typename = void>
+struct laplace_options;
 
-  /* Initial value for theta. Defaults to 0s of the correct size if nullopt */
-  std::optional<Eigen::VectorXd> theta_0{std::nullopt};
+template <typename Theta>
+struct laplace_options<Theta, require_eigen_t<Theta>> : public laplace_options_base {
+  /* Value for user supplied initial theta  */
+  Theta theta_0{0};
 };
 
+template <typename Theta>
+struct laplace_options<Theta, require_not_eigen_t<Theta>> : public laplace_options_base {};
+
+using laplace_options_default = laplace_options<void>;
```

Comment:
Would it be possible just to have this be `template <bool HasInitTheta>` and then the two specializations just being `false` and `true`? 

Something like 

```c++
template <bool HasInitTheta>
struct laplace_options;

template <>
struct laplace_options<false> : public laplace_options_base {};

template <>
struct laplace_options<true> : public laplace_options_base {
  /* Value for user supplied initial theta  */
  Eigen::VectorXd theta_0{0};
};


using laplace_options_default = laplace_options<false>;
using laplace_options_theta0 = laplace_options<true>;

```

seems cleaner to me than require/require_nots and a `void`

Reply:
I like it! Just updated